Userland/Libraries/LibCore/DateTime.cpp

```diff
@@ -85,6 +85,16 @@ void DateTime::set_time(int year, int month, int day, int hour, int minute, int
     m_second = tm.tm_sec;
 }
 
+void DateTime::set_time_only(int hour, int minute, int second)
+{
+    set_time(year(), month(), day(), hour, minute, (second == -1) ? this->second() : second);
+}
+
+void DateTime::set_date(Core::DateTime other)
+{
+    set_time(other.year(), other.month(), other.day(), hour(), minute(), second());
+}
+
```

Comment:
A couple of points on this:
- These DateTime changes are distinct, so they belong in a separate commit before this one.
- Please use `Optional<int>` instead of giving `-1` a special meaning. (You can use `{}` as the default value to mean an empty Optional, and then something like `second.value_or(this->second())` to get the value out.)
- `DateTime::set_date()` should take a const reference to `other` instead of copying it.